src/scitokens_internal.cpp

```diff
@@ -1064,3 +1065,75 @@ bool scitokens::Enforcer::scope_validator(const jwt::claim &claim,
 
     return me->m_test_authz.empty();
 }
+
+// Configuration class functions
+std::pair<bool, std::string>
+configurer::Configuration::set_cache_home(const std::string dir_path) {
+    // If setting to "", then we should treat as though it is unsetting the
+    // config
+    if (dir_path.length() == 0) { // User is configuring to empty string
+        m_cache_home = std::make_shared<std::string>(dir_path);
+        return std::make_pair(true, "");
+    }
+
+    // Check that the cache_home exists, and if not try to create it
+    if (!check_dir(dir_path)) {
+        if (!mkdir_and_parents_if_needed(dir_path)) {
+            return std::make_pair(false, "The provided cache home path did not "
+                                         "exist and could not be created.");
+        }
+    }
+
+    // Now it exists and we can write to it, set the value and let
+    // scitokens_cache handle the rest
+    m_cache_home = std::make_shared<std::string>(dir_path);
+    return std::make_pair(true, "");
+}
+
+std::string configurer::Configuration::get_cache_home() { return *m_cache_home; }
+
+bool configurer::Configuration::check_dir(const std::string dir_path) {
+    struct stat info;
+    return stat(dir_path.c_str(), &info) == 0 && (info.st_mode & S_IFDIR);
+}
+
+bool configurer::Configuration::mkdir_and_parents_if_needed(
+    const std::string dir_path) {
+    // SciTokens-cpp already makes assumptions about using Linux file paths,
+    // so making that assumption here as well.
+
+    // Using these perms because that's what the actual cache file uses in
+    // scitokens_cache
+    mode_t mode = 0700; // Maybe these permissions should be configurable?
+
+    int result;
+    std::string currentLevel;
+    std::vector<std::string> pathComponents = path_split(dir_path);
+    for (const auto &component : pathComponents) {
+        currentLevel += "/" + component;
+        if (!check_dir(currentLevel)) {
```

Comment:
Don't check the dir first; just create it and inspect the `mkdir` return code (if it's `EEXIST`, then the directory already exists and move on).

For error message creation, it would be better to have an output variable that's the error message so you can say precisely the file operation that failed.  The caller of this function currently has something generic that won't be useful for debugging.